graphql/resolvers/exerciseCrud.ts

```diff
@@ -91,3 +95,75 @@ export const deleteExercise = async (
     }
   })
 }
+
+export const flagExercise = async (
+  _parent: void,
+  arg: MutationFlagExerciseArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id, flagReason } = arg
+
+  const flaggedById = req.user?.id
+  if (!flaggedById) throw new Error('No User')
+
+  const exercise = await prisma.exercise.findUnique({
+    where: {
+      id
+    }
+  })
+
+  if (exercise?.flaggedAt) {
+    throw new Error('Exercise is already flagged')
+  }
+
+  return prisma.exercise.update({
+    where: { id },
+    data: {
+      flaggedAt: new Date().toISOString(),
+      flaggedById,
+      flagReason
+    },
+    include: {
+      author: true,
+      module: true
+    }
+  })
+}
+
+export const removeExerciseFlag = async (
+  _parent: void,
+  arg: MutationRemoveExerciseFlagArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id } = arg
+
+  const adminId = req.user?.id
+  if (!adminId) throw new Error('No User')
```

Comment:
Now that they are side by side, I realize lines 140 and 140 is unnecessary because of line 143. Basically 147 only hits if user isAdmin, and if user is admin there is 100% chance `req.user.id` exists: https://github.com/garageScript/c0d3-app/blob/master/prisma/schema.prisma#L140